walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -15,9 +15,124 @@
+                end = occasion.getWhen().start();
+                if(meetingFits(request.getDuration(), start, end)) {
+                    workingTimes.add(TimeRange.fromStartEnd(start, end, false));
+                }
+                // Change start if endpoint current person's occasion is greater
+                if(occasion.getWhen().end() > start) {
+                    start = occasion.getWhen().end();
+                }
+                foundGuest = false;
+            }
+        }
+        workingTimes = updateWithEndOfDay(workingTimes, start, request.getDuration());
+        workingTimes = updateWithOptionalAttendees(workingTimes, events, request);
+        return workingTimes;
+    }
+
+    /** Searches through the optional attendees and attempts to place them in time blocks
+     *  that work for mandatory attendees. Modifies the Collection of TimeRanges if so.
+    */
+    private Collection<TimeRange> updateWithOptionalAttendees(Collection<TimeRange> workingTimes, 
+                                                                Collection<Event> events, 
+                                                                MeetingRequest request) {
+        boolean foundGuest = false;
+        for(Event occasion: events) {
+            for(String guest: request.getOptionalAttendees()) {
+                if(occasion.getAttendees().contains(guest)) {
+                    foundGuest = true;
+                    break;
+                }
+            }
+            if(foundGuest) {
+                int start = occasion.getWhen().start();
+                int end = occasion.getWhen().end();
+                boolean isAtEndOfDay = (end == TimeRange.END_OF_DAY)? true: false;
+                TimeRange guestMeeting = TimeRange.fromStartEnd(start, end, isAtEndOfDay);
+                for(TimeRange currentTimeRange: workingTimes) {
+                    if(currentTimeRange.contains(guestMeeting)){
+                        workingTimes = splitTimeRange(workingTimes, currentTimeRange, 
+                                                        start, end, request.getDuration());
+                        break;
+                    }
+                }
+                foundGuest = true;
+            }
+        }
+        return workingTimes;
+    }
+
+    /** Handles the cases where a new time block may be open before or after the meeting. 
+     *  Determines if the current TimeRange should be removed and update workingTimes.
+     */
+    private Collection<TimeRange> splitTimeRange(Collection<TimeRange> workingTimes, 
+                                                TimeRange candidate, int start, 
+                                                int end, long duration) {
+        int newStartTime = candidate.start();
+        int newEndTime = candidate.end();
+        boolean remove = false;
+
+        // Check block of time before meeting
+        if(newStartTime != start && meetingFits(duration, newStartTime, start)) {
```

Comment:
These if statements seem to be mutually exclusive to each other. If the code will only ever trigger one of the conditionals, you should use chained `else if` statements rather than new `if` statements. This makes it more obvious that the conditionals are mutually exclusive conditions.